src/providers/cryptoauthlib/generate_random.rs

```diff
@@ -0,0 +1,46 @@
+use super::Provider;
+use parsec_interface::operations::psa_generate_random;
+use parsec_interface::requests::{ResponseStatus, Result};
+use zeroize::Zeroizing;
+
+impl Provider {
+    pub(super) fn psa_generate_random_internal(
+        &self,
+        op: psa_generate_random::Operation,
+    ) -> Result<psa_generate_random::Result> {
+        let mut random_bytes = vec![0u8, 0];
+        // check input size
+        // size 0 - no need to run operation
+        if op.size == 0 {
+            Ok(psa_generate_random::Result {
+                random_bytes: Zeroizing::new(vec![]),
+            })
+        }
+        // for size bigger than buffer size operation need to be executed multiple times
+        else {
+            // calculate loop count
+            let call_count = (op.size / rust_cryptoauthlib::ACTA_RANDOM_BUFFER_SIZE) + 1;
```

Comment:
Forgot to put it in the review 🤦 

Sorry to nerd out, but I think you can remove the `if`/`else` if you calculate `call_count` as:
```rust
let call_count = (op.size + rust_cryptoauthlib::ACTA_RANDOM_BUFFER_SIZE - 1) / rust_cryptoauthlib::ACTA_RANDOM_BUFFER_SIZE;
```

Also a bit more efficient for when `op.size` is a multiple of `rust_cryptoauthlib::ACTA_RANDOM_BUFFER_SIZE` 🧠

Reply:
? if `op.size` is equal to `rust_cryptoauthlib::ACTA_RANDOM_BUFFER_SIZE`, then your formula sets `call_count` to 1, whereas the existing one sets it to 2